### DISCUSSION TITLE
Problems running MintPy with HyP3

### DISCUSSION THREAD

**Discussion by anaferreira97** [insarlab/MintPy#790] [Q&A]
Hello everyone!

I'm trying to run Mintpy with HyP3 processor but I'm having problems generating temporal coherence mask. The log file is attached.
I don't know why I'm not having enough reliable pixels. All my clip files have at least 80MB. Do you have any idea why? 

[logfile.txt](https://github.com/insarlab/MintPy/files/8892107/logfile.txt)

**Comment by mosyhey** [comment]
Three interferograms have been dropped from processing due to not having the correct size. First, fix this warning by means of `cutGeotiffs.py`, as explained [here](https://github.com/ASFHyP3/hyp3-lib/issues/263).
```
********************************************************************************
WARNING: NOT all input unwrapped interferograms have the same row/column number!
The most common size is: (5769, 4707)
------------------------------
The following dates have different size:
	201223-210104	(5607, 4707)
	201223-210116	(5607, 4707)
	210104-210116	(5769, 4705)
------------------------------
Skip loading the above interferograms (3).
Continue to load the rest interferograms (2).
********************************************************************************
```

**Reply by jhkennedy** [reply]
Yes, it looks like some have been dropped for being the wrong size. @anaferreira97 how did you prepare the HyP3 data for MintPy?

@mosyhey in general, we (the HyP3 development team), no longer recommend using `hyp3lib` outside of developing HyP3 plugins -- much of the code is untested and was written for GDAL v1. We've created an updated tutorial for preparing HyP3 products here: <https://hyp3-docs.asf.alaska.edu/tutorials/mintpy>. It includes downloading and unzipping HyP3 products, clipping them to their common overlap, and writing the MintPy config file.

**Reply by mosyhey** [reply]
Thank you dear @jhkennedy.
It was great help and very useful notebook.

**Reply by anaferreira97** [reply]
Hi everyone! Thank you for your help.
I already try using the notebook but the error still occurs... I don't understand what can I be doing wrong...

**Reply by anaferreira97** [reply]
The error that still occurs is:
`RuntimeError: Not enough reliable pixels (minimum of 100). Try the following:
1) Check the reference pixel and make sure it's not in areas with unwrapping errors
2) Check the network and make sure it's fully connected without subsets`

**Reply by anaferreira97** [reply]
[logfile.txt](https://github.com/insarlab/MintPy/files/9139509/logfile.txt)

**Reply by yunjunz** [reply]
To better diagnose, could you try:
1. update your mintpy to the latest development version, and re-run?
2. upload the `pic` folder for us to see the data quality?

The recordings from previous years' UNAVCO course (https://github.com/insarlab/mintpy-tutorial#1-small-baseline-time-series-analysis-with-smallbaselineapppy) can be helpful, if you are familiar with what each step is doing or checking.

**Reply by anaferreira97** [reply]
Hi yunjunz! Thank you so much for your feedback.

I already re-run the process and the error still occurs. My pic folder only have the txt and the config file...

**Reply by mosyhey** [reply]
Hi. As you can see in the generated file named `network.pdf`, the interferogram network is very poor and not fully connected:
```
***WARNING: the network is NOT fully connected.
	Inversion result can be biased!
	Continue to use SVD to resolve the offset between different subsets.
-------------------------------------------------------------------------------
number of interferograms: 12
number of acquisitions  : 14
```
fix this warning too, and rerun the process.
![poor_rich_network](https://user-images.githubusercontent.com/105506772/183277719-a235bbdb-8111-444e-99d9-eed4caefdb4a.png)

**Reply by yunjunz** [reply]
I second @mosyhey comment above. 

You could run `smallbaselineApp.py --plot` to populate the `pic` folder. I will turn ON this behavior by default in the next commit (update: this is now implemented in https://github.com/insarlab/MintPy/pull/825).

**Comment by AuRevoirzzz** [comment]
I encountered the same issue, but I was using ISCE + MintPy. Have you resolved this problem now?
[logfile.txt](https://github.com/insarlab/MintPy/files/15131511/logfile.txt)
